src/content/guides/packages.md

```diff
@@ -178,6 +178,78 @@ due to conflicting version constraints in the pubspec.
 To identify out-of-date packages that require editing the pubspec,
 use [`dart pub outdated`][outdated].
 
+## Getting dependencies in production - `pub get --enforce-lockfile`
+
+In some cases `dart pub get` will not give you the exact versions locked in
+`pubspec.lock`. For example:
+
+* If new dependencies are added to `pubspec.yaml` since `pubspec.lock` was created.
+* If the locked version no longer exists in the package repository.
+* If you upgraded the Dart SDK, and some packages no longer are compatible with
+  the sdk.
```

Comment:
Assuming this can also happen on downgrading SDK versions:

```suggestion
* If you changed to a different version of the Dart SDK,
  and some packages are no longer compatible with that new version.
```